app/lib/efile/md/md502_r_calculator.rb

```diff
@@ -0,0 +1,35 @@
+module Efile
+  module Md
+    class Md502RCalculator < ::Efile::TaxCalculator
+      attr_reader :lines, :value_access_tracker
+
+      def initialize(value_access_tracker:, lines:, intake:)
+        @value_access_tracker = value_access_tracker
+        @lines = lines
+        @intake = intake
+        @direct_file_json_data = intake.direct_file_json_data
+      end
+
+      def calculate
+        set_line(:MD502R_LINE_9A, :calculate_line_9a)
+        set_line(:MD502R_LINE_9B, :calculate_line_9b)
+      end
+
+      private
+
+      def calculate_line_9a
+        if @intake.filing_status_mfj? && @intake.direct_file_data.fed_ssb.positive? && @intake.primary_ssb_amount.present?
+          @intake.primary_ssb_amount.round
+        else
+          @intake.direct_file_data.fed_ssb.round
```

Comment:
what if `primary_ssb_amount` is 0 but `spouse_ssb_amount` is positive? do we want to round the `fed_ssb` for this scenario? Also, should we return 0 or nil here?

I'd suggest:
```
    if @intake.direct_file_data.fed_ssb.positive?
      if @intake.filing_status_mfj?
        @intake.primary_ssb_amount&.round || 0
      else
        @intake.direct_file_data.fed_ssb.round
      end
    end
```